src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,426 @@
+                final List<String> firstNameMatches = firstNameSet.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+                if (!firstNameMatches.isEmpty())
+                {
+                    return firstNameMatches;
+                }
+            }
+        }
+        else
+        {
+            if (1 == currentWordIndex)
+            {
+                final Set<String> middleNameSet = new HashSet<>();
+                for (final String citizenName : citizenNameStrings)
+                {
+                    final String[] citizenNameParts = citizenName.split(" ");
+                    if (potentiaCitizenNameParts[0].equals(citizenNameParts[0]))
+                    {
+                        middleNameSet.add(citizenNameParts[1]);
+                    }
+                }
+                return middleNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[1])).collect(Collectors.toList());
+            }
+            else
+            {
+                final Set<String> lastNameSet = new HashSet<>();
+                for (final String citizenName : citizenNameStrings)
+                {
+                    final String[] citizenNameParts = citizenName.split(" ");
+                    if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
+                    {
+                        lastNameSet.add(citizenNameParts[2]);
+                    }
+                }
+                return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
+           }
+        }
+        return Collections.emptyList();
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
+
+        switch (this)
+        {
+            case INTEGER:
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return Ints.tryParse(potentialArgumentValue);
+            case BOOLEAN:
+                return parseBoolean(potentialArgumentValue);
+            case PLAYER:
+                return parsePlayerValue(server, potentialArgumentValue);
+            case COLONY:
+                return parseColonyValue(sender, potentialArgumentValue);
+            case CITIZEN:
+                return parseCitizenDataValue(parsedHolders, potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType parsing");
+        }
+    }
+
+    private Boolean parseBoolean(final String potentialArgumentValue)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make "parseBoolean" a "static" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2325)